### PR TITLE
feat: Adding the integer conversion operations to the llvm dialect

### PR DESCRIPTION
**Pull request by lfrenot**

### CODE REVIEW

**Comment by tobiasgrosser** [comment]
Nice. I guess we need to fix ci and maybe drop the commented icmp stuff.

File: SSA/Projects/InstCombine/LLVM/PrettyEDSL.lean [hunk 1]
```diff
@@ -145,6 +157,14 @@ private def pretty_test_overflow :=
     llvm.return %3 : i32
   }]
 
+private def prettu_test_trunc :=
+  [llvm ()|{
```
Comment (tobiasgrosser) [inline SSA/Projects/InstCombine/LLVM/PrettyEDSL.lean]:
```suggestion
private def pretty_test_trunc :=
  [llvm ()|{
```

File: SSA/Projects/InstCombine/LLVM/EDSL.lean [hunk 2]
```diff
@@ -161,6 +161,11 @@ def mkExpr (Γ : Ctxt (MetaLLVM φ).Ty) (opStx : MLIR.AST.Op φ) :
         | "llvm.icmp.sge" => pure <| Sum.inr LLVM.IntPredicate.sge
         | "llvm.icmp.slt" => pure <| Sum.inr LLVM.IntPredicate.slt
         | "llvm.icmp.sle" => pure <| Sum.inr LLVM.IntPredicate.sle
+        -- | "llvm.icmp" => do
+        --   let attr? := opStx.attrs.getAttr "predicate"
+        --   match attr? with
+        --     | .none => throw <| .generic s!"What?"
+        --     | .some y => throw <| .generic s!"Predicate flag: {MLIR.AST.docAttrVal y}."
```
Comment (tobiasgrosser) [inline SSA/Projects/InstCombine/LLVM/EDSL.lean]: You probably want to drop this in this PR and work on it in a different PR, no?

Comment (lfrenot) [reply SSA/Projects/InstCombine/LLVM/EDSL.lean]: Oops, yes

**Review by tobiasgrosser — APPROVED** [review]

**Pull request merged**